crypto/bio/bss_dgram_pair.c

```diff
@@ -745,7 +745,9 @@ static size_t dgram_pair_read_inner(struct bio_dgram_pair_st *b, uint8_t *buf, s
 
         ring_buf_pop(&b->rbuf, src_len);
 
-        buf         += src_len;
+        if (buf != NULL)
```

Comment:
(v minor) might be a bit nicer to move this into the existing `if (buf != NULL)` check at line 743